### ISSUE TITLE
custom url not queued if internet is not available (no signal)

### ISSUE THREAD

**Issue by damgut** [mendhak/gpslogger#795]
Using custom url to update the position to a server is queued if the phone is set in offline mode. --> OK

But if the phone is online and the host is not reached (for example, no signal), the request will be repeated every 10 seconds 5 times and after that the position is not updated any more. It would be nice to use the same procedure to queue and try to send forever. The pause to retry could be the same as the actualization time configured.

**Comment by algor512** [comment]
I also struggled with this issue. I use gpslogger also to track family members on long-distance trips, the signal during these trips is not always stable, and it is quite important to send a position once there is a signal as soon as possible.

**Comment by mendhak** [comment]
Yes the Custom URL will give up after a while, the reason is to avoid having it try forever, or it would become a battery drain.  The Custom URL _sender_ feature was created for this purpose though, it can be configured to re-send all the previous points once every X minutes.  It's still subject to the maximum 5 attempts, but the re-sending is on a schedule, so there are more attempts.

**Comment by mendhak** [comment]
Only thing I have is https://gpslogger.app/  
If there's some improvements that can be made to that structure let me know, I can try.  Maybe I just need a tour of each screen at the top with a little detail?

**Comment by MrColumbo** [comment]
> Yes the Custom URL will give up after a while, the reason is to avoid having it try forever, or it would become a battery drain. The Custom URL _sender_ feature was created for this purpose though, it can be configured to re-send all the previous points once every X minutes. It's still subject to the maximum 5 attempts, but the re-sending is on a schedule, so there are more attempts.

Is it possible to configure how soon it gives up? I am thinking that i rather like the app to send data only ever 4 hours and rather give up quite quickly when it tries to send the points. My main goal is to make sure data is uploaded (at a later time) when hiking somewhere where there is not cellphone coverage. Actually i could even imagine that telling the app to only upload data when it is getting charged. I guess if there is a decent data connection it will be easy to upload a batch of maybe even a thousand points and it might be worth to let the app wait until it is possible to send it via a decent connection. By the way .....i have setup autosend to do it every 60 minutes but in fact the app is sending data in real time. Does the "How often?" setting work for custom url? I will try to configure it to only send via WLAN .....that might be a good workaround.

**Comment by mendhak** [comment]
It's not possible to configure that. The auto send feature better suited for the 'reliable' sending as it retries sending at configurable minutes. You can also set autosending in general to use Wifi and that will in turn make all network communication wait for Wifi to be available. 

>  By the way .....i have setup autosend to do it every 60 minutes but in fact the app is sending data in real time.

If you go into custom URL settings it's possible both "log to custom URL" and "allow auto sending" are enabled. Disable the "log to custom URL" and leave the "allow auto sending" enabled, and that should send it every 60 minutes.
